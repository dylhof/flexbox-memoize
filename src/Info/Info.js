import React from 'react';

function Info(props) {
    return (
      <div className='info-div'>
        {!props.quizStarted && 
          <div className='aside-instruct-div'>
            <p>Instructions</p>
            <ol>
              <li>Click StartQuiz</li>
              <li>Choose the image that may be produced if the named flexbox property were in your CSS</li>
              <li>If you get stuck, check back here to get more information about the flexbox property</li>
              <li>Enjoy creating simple layouts with flexbox!</li>
            </ol>
          </div>
        }
        {props.quizStarted && !props.showInfo && 
            <p className='show-more-info-p' onClick={props.toggleShowInfo}>
              Click Here to show more info about: 
              <span>{props.currentQuestion.property}</span> 
            </p>    
        }
        {props.quizStarted && props.showInfo && 
            <p className='property-info-p' onClick={props.toggleShowInfo}>
              {props.currentQuestion.property}
              {props.currentQuestion.info}
            </p>
        }
      </div>
    )
}

export default Info;