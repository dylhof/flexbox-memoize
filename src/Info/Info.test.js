import React from 'react';
import Info from './Info';
// import Instructions from '../Instructions/Instuructions.js'
import { shallow } from 'enzyme';

const question = {
  "property": "flex-direction: row",
  "correctAnswer": "https://i.imgur.com/Xo1MD86.png",
  "answers": ["https://i.imgur.com/Xo1MD86.png", "https://i.imgur.com/JIJHnVf.png", "https://i.imgur.com/4y9cZDw.png", "https://i.imgur.com/Id0kLlR.png"],
  "info": "This establishes the main-axis, thus defining the direction flex items are placed in the flex container. Using row organizes the items from left to right.",
  "family-member": "Parent",
  "model": "Flexbox"
}
const toggleShowInfoMock= jest.fn();

describe('Info', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Info 
        currentQuestion={question}
        showInfo={false}
        quizStarted={true}
        toggleShowInfo={toggleShowInfoMock}/>
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot', () => {
    let wrapper = shallow(
      <Info 
        currentQuestion={question}
        showInfo={false}
        quizStarted={false}/>
    )
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot', () => {
    let wrapper = shallow(
      <Info 
        currentQuestion={question}
        showInfo={true}
        quizStarted={false}/>
    )
    expect(wrapper).toMatchSnapshot();
  })

  it('should call toggleShowInfo when clicked', () => {
    wrapper.find('.show-more-info-p').simulate('click');
    expect(toggleShowInfoMock).toBeCalled();
  })
})