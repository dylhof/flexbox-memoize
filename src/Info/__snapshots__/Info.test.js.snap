// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info should 1`] = `
<div>
  <div>
    <p>
      Instructions
    </p>
    <ol>
      <li>
        Click StartQuiz
      </li>
      <li>
        Choose the image that may be produced if the named flexbox property were in your CSS
      </li>
      <li>
        If you get stuck, check back here to get more information about the flexbox property
      </li>
      <li>
        Enjoy creating simple layouts with flexbox!
      </li>
    </ol>
  </div>
</div>
`;

exports[`Info should match the snapshot 1`] = `
<div>
  <p>
    Click Here to show more info about:
    <span>
      flex-direction: row
    </span>
  </p>
</div>
`;

exports[`Info should match the snapshot 2`] = `
<div>
  <div>
    <p>
      Instructions
    </p>
    <ol>
      <li>
        Click StartQuiz
      </li>
      <li>
        Choose the image that may be produced if the named flexbox property were in your CSS
      </li>
      <li>
        If you get stuck, check back here to get more information about the flexbox property
      </li>
      <li>
        Enjoy creating simple layouts with flexbox!
      </li>
    </ol>
  </div>
</div>
`;
